dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldCriteria.java

```diff
@@ -0,0 +1,164 @@
+package com.dotcms.contenttype.business.uniquefields.extratable;
+
+import com.dotcms.content.elasticsearch.business.ESContentletAPIImpl;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.variant.model.Variant;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.util.StringUtils;
+import com.liferay.util.StringPool;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represent the criteria used to determine if a value is unique or not
+ */
+class UniqueFieldCriteria {
+    private final ContentType contentType;
+    private final Field field;
+    private final Object value;
+    private final Language language;
+    private final Host site;
+    private String variantName;
+
+    public UniqueFieldCriteria(final Builder builder) {
+        this.contentType = builder.contentType;
+        this.field = builder.field;
+        this.value = builder.value;
+        this.language = builder.language;
+        this.site = builder.site;
+        this.variantName = builder.variantName;
+    }
+
+
+    /**
+     * Return a Map with the values in this Unique Field Criteria
+     * @return
+     */
+    public Map<String, Object> toMap(){
+        final Map<String, Object> map = new HashMap<>(Map.of(
+                "contentTypeID", Objects.requireNonNull(contentType.id()),
+                "fieldVariableName", Objects.requireNonNull(field.variable()),
+                "fieldValue", value.toString(),
+                "languageId", language.getId(),
+                "uniquePerSite", isUniqueForSite(contentType.id(), field.variable()),
+                "variant", variantName
+        ));
+
+        if (site != null) {
+            map.put("hostId", site.getIdentifier());
+        }
+
+        return map;
+    }
+
+    /**
+     * return true if the uniquePerSite Field Variable is set to true.
+     *
+     * @param contentTypeId
+     * @param fieldVariableName
+     * @return
+     */
+    private static boolean isUniqueForSite(String contentTypeId, String fieldVariableName)  {
+        try {
+            final Field uniqueField = APILocator.getContentTypeFieldAPI().byContentTypeIdAndVar(contentTypeId, fieldVariableName);
+
+            return uniqueField.fieldVariableValue(ESContentletAPIImpl.UNIQUE_PER_SITE_FIELD_VARIABLE_NAME)
+                    .map(Boolean::valueOf).orElse(false);
+        } catch (DotDataException e) {
+            throw  new DotRuntimeException(e);
```

Comment:
Let's add both the contant type ID and field var name to the error message

Reply:
done https://github.com/dotCMS/core/pull/30466/files#diff-6d5733d9cc36530da02b28339a7097d0d4b8ee3e94ea537001d91bfa28cf4864R76